rust/core-lib/src/rpc.rs

```diff
@@ -533,10 +533,11 @@ impl<'de, T: Deserialize<'de>> Deserialize<'de> for EditCommand<T>
         let mut v = Value::deserialize(deserializer)?;
         let helper = InnerId::deserialize(&v).map_err(de::Error::custom)?;
         let InnerId { view_id } = helper;
+
         // if params are empty, remove them
         let remove_params = match v.get("params") {
-            Some(&Value::Object(ref obj)) => obj.is_empty(),
-            Some(&Value::Array(ref arr)) => arr.is_empty(),
+            Some(&Value::Object(ref obj)) => obj.is_empty() && T::deserialize(v.clone()).is_err(),
+            Some(&Value::Array(ref arr)) => arr.is_empty() && T::deserialize(v.clone()).is_err(),
```

Comment:
I think this is an improvement, but it still means we're going to be trying and failing to deserialize an awful lot of RPCs, and I _am_ curious about the performance implications.

Reply:
Good question, I'll look into this later today.

Reply:
Interesting. I measured the deserialization performance for `find_next` with and without parameters. There is basically no difference. 
I measured the time using `xi_trace` from line 537 until after line 549. With and without parameters it takes ~ 0.050ms.

Reply:
Okay, thanks for measuring!